test/dash-playlist-loader.test.js

```diff
@@ -30,51 +30,149 @@ QUnit.test('throws if the playlist url is empty or undefined', function(assert)
+QUnit.skip('moves to HAVE_MASTER after loading a master playlist', function(assert) {
   let loader = new DashPlaylistLoader('dash.mpd', this.fakeHls);
+  let state;
+
+  loader.on('loadedplaylist', function() {
+    state = loader.state;
+  });
 
   loader.load();
-  assert.equal(this.requests.length, 1, 'made a request');
-  assert.equal(this.requests[0].url, 'dash.mpd', 'requested the manifest');
+  standardXHRResponse(this.requests.shift());
+
+  assert.ok(loader.master, 'the master playlist is available');
+  assert.strictEqual(state, 'HAVE_MASTER', 'the state at loadedplaylist correct');
 });
 
-QUnit.test('moves to HAVE_MASTER and HAVE_METADATA after loading the manifest',
-function(assert) {
+QUnit.test('moves to HAVE_METADATA after loading a media playlist', function(assert) {
+  let loadedPlaylist = 0;
+  let loadedMetadata = 0;
   let loader = new DashPlaylistLoader('dash.mpd', this.fakeHls);
-  let loadedPlaylistStates = [];
 
   loader.load();
 
   loader.on('loadedplaylist', function() {
-    loadedPlaylistStates.push(loader.state);
+    loadedPlaylist++;
+  });
+  loader.on('loadedmetadata', function() {
+    loadedMetadata++;
   });
+
   standardXHRResponse(this.requests.shift());
-  assert.ok(loader.master, 'the master playlist is available');
+  this.clock.tick(1);
+
   // because DASH only has one manifest, it should go through two loadedplaylists
   // and end with HAVE_METADATA because it already has the first media ready
-  assert.equal(loadedPlaylistStates.length, 2, 'triggered two loadedplaylist events');
-  assert.equal(loadedPlaylistStates[0], 'HAVE_MASTER', 'got master first');
-  assert.equal(loadedPlaylistStates[1], 'HAVE_METADATA', 'got media second');
+  assert.strictEqual(loadedPlaylist, 2,
+    'fired loadedplaylist once for master playlist and once for' +
+    ' autoselected first child playlist');
+  // this is triggered on the next tick after construction
+  assert.strictEqual(loadedMetadata, 1, 'fired loadedmetadata once');
+  assert.ok(loader.master, 'sets the master playlist');
+  assert.ok(loader.media(), 'sets the media playlist');
+  assert.strictEqual(this.requests.length, 0, 'no further requests are needed');
+
+  const childLoader = new DashPlaylistLoader(loader.master.playlists[0], this.fakeHls, false, loader);
+
+  childLoader.on('loadedplaylist', function() {
+    loadedPlaylist++;
+  });
+  childLoader.on('loadedmetadata', function() {
+    loadedMetadata++;
+  });
+
+  childLoader.load();
+  this.clock.tick(1);
+
+  // actually, loadedplaylist is triggered 4 times at this point,
+  // but the 3rd is on childLoader's creation so it's not observable
+  assert.strictEqual(loadedPlaylist, 3, 'fired loadedplaylist thrice');
+  assert.strictEqual(loadedMetadata, 2, 'fired loadedmetadata twice');
+  assert.strictEqual(loader.state, 'HAVE_METADATA', 'the state is correct');
+});
+
+QUnit.test('jumps to HAVE_METADATA when initialized with a live media playlist',
```

Comment:
I don't understand the point of this test. This is exactly the behavior we do not want.

Reply:
I think this will probably get reworked or replaced with the skipped test once the code changes are in

Reply:
I think this should probably actually be for a child playlist loader instead of how I have it right now

Reply:
I thought the point of these added tests was to write tests that test the functionality we want in this new DashPlaylistLoader world, if this test is not valid in that world, we should remove it. This test being for the child playlist loader makes sense. However, is it really testing anything different from the test above?